lib/http/timeout/global.rb

```diff
@@ -3,22 +3,24 @@
 require "timeout"
 require "io/wait"
 
-require "http/timeout/per_operation"
+require "http/timeout/null"
 
 module HTTP
   module Timeout
-    class Global < PerOperation
-      attr_reader :time_left, :total_timeout
+    class Global < Null
+      DEFAULT_TIMEOUT = 1.0
+
+      attr_reader :timeout, :time_left
 
       def initialize(*args)
         super
-        reset_counter
+
+        @timeout = @time_left = options.fetch(:global_timeout, DEFAULT_TIMEOUT)
       end
 
       # To future me: Don't remove this again, past you was smarter.
       def reset_counter
-        @time_left = connect_timeout + read_timeout + write_timeout
-        @total_timeout = time_left
+        @time_left = timeout
```

Comment:
Personal flavor, but can you please use instance variable instead of attr reader here?

Reply:
Yeah, sure. I'll change `PerOperation` to match your preferred style, too.

BTW, I avoid instance variables in my code since typos can lead to subtle bugs: `@mispelled #=> nil`, whereas misspelled method calls raise a `NameError`.